web_services/foxx/brightid/db.js

```diff
@@ -319,168 +166,145 @@ function invite(inviter, invitee, groupId, data, timestamp) {
-  addUserToGroup(groupId, key1, timestamp);
-  invite(key1, key2, groupId, inviteData2, timestamp);
-  invite(key1, key3, groupId, inviteData3, timestamp);
+  if (type == 'family') {
+    const fg = familyGroup(key);
+    if (fg && fg.head == key) {
+      throw new errors.AlreadyIsFamilyHead();
+    }
+    group.head = key;
+    group.vouchers = [];
+  }
+
+  groupsColl.insert(group);
+  // Add the creator to the group now.
+  addUserToGroup(groupId, key, timestamp);
 }
 
 function addAdmin(key, admin, groupId) {
-  if (! groupsColl.exists(groupId)) {
-    throw new errors.GroupNotFoundError(groupId);
+  const group = getGroup(groupId);
+  if (! group.admins || ! group.admins.includes(key)) {
+    throw new errors.NotAdminError();
   }
   if (! usersInGroupsColl.firstExample({
-    _from: 'users/' + admin,
-    _to: 'groups/' + groupId
-  })) {
+      _from: 'users/' + admin,
+      _to: 'groups/' + groupId
+    })) {
     throw new errors.IneligibleNewAdminError();
   }
-  const group = groupsColl.document(groupId);
-  if (! group.admins || ! group.admins.includes(key)) {
-    throw new errors.NotAdminError();
-  }
   group.admins.push(admin);
   groupsColl.update(group, { admins: group.admins });
 }
 
 function addUserToGroup(groupId, key, timestamp) {
-  const user = 'users/' + key;
-  const group = 'groups/' + groupId;
+  const _from = 'users/' + key;
+  const _to = 'groups/' + groupId;
 
-  const edge = usersInGroupsColl.firstExample({
-    _from: user,
-    _to: group
-  });
+  const edge = usersInGroupsColl.firstExample({ _from, _to });
   if (! edge) {
-    usersInGroupsColl.insert({
-      _from: user,
-      _to: group,
-      timestamp
-    });
+    usersInGroupsColl.insert({ _from, _to, timestamp });
   } else {
     usersInGroupsColl.update(edge, { timestamp });
   }
-
-}
-
-function addMembership(groupId, key, timestamp) {
-  if (! groupsColl.exists(groupId)) {
-    throw new errors.GroupNotFoundError(groupId);
-  }
-
+  // empty the group's vouchers after family group member changes
   const group = groupsColl.document(groupId);
-  if (group.isNew && ! group.founders.includes(key)) {
-    throw new errors.NewUserBeforeFoundersJoinError();
+  if (group.type == 'family') {
+    groupsColl.update(group, { vouchers: [] });
   }
+}
 
-  if (group.type == 'primary' && hasPrimaryGroup(key)) {
-    throw new errors.AlreadyHasPrimaryGroupError();
+function knownConnections(userId) {
+  return query`
+    FOR conn in ${connectionsColl}
+      FILTER conn._from == ${'users/' + userId}
+      FILTER conn.level IN ['already known', 'recovery']
+      FOR conn2 in ${connectionsColl}
+        FILTER conn2._from == conn._to
+        FILTER conn2._to == conn._from
+        FILTER conn2.level IN ['already known', 'recovery']
+      RETURN conn._to
+  `.toArray().map(id => id.replace('users/', ''));
+}
+
+function checkJoiningFamily(groupId, userId) {
+  const fg = familyGroup(userId);
+  if (fg && fg.head != userId) {
```

Comment:
Once the `familyGroup` function is fixed this can be changed to just
```
if (familyGroup(userId))
```
because it will return the one family group for the user (where they are not the head).

Reply:
Done